src/PrestaShopBundle/Resources/config/services/core/grid/doctrine_query_builder.yml

```
@@ -179,7 +179,7 @@ services:
     parent: 'prestashop.core.grid.abstract_query_builder'
     arguments:
       - '@prestashop.core.query.doctrine_search_criteria_applicator'
-      - "@=service('prestashop.adapter.legacy.context').getContext().employee.id_lang"
+      - "@=service('prestashop.default.language.context')"
```

```suggestion
      - "@PrestaShop\PrestaShop\Core\Context\LanguageContext"
```
Prefer the simple notation and ideally the FQCN, you could even add the `autowire: true` option if you want.

Besides, be careful the `prestashop.default.language.context` will not return the Employee contextual language, it is used to have the Shop's default language, which is not what you want in this case.

- `PrestaShop\PrestaShop\Core\Context\LanguageContext` is a "dynamic context", it uses the Employee custom language in the BO, in API context it is built differently (either default language by default, or overriden via a query parameter)
- `prestashop.default.language.context` is a bit "more static" and is always filled with the default language configured in the configuration DB

---

I hadn't even noticed this small detail – but it really makes a difference!
You don't miss a thing! 😄

---

@jolelievre I've applied the changes, opting for constructor property promotion and `autowire: true`.